fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1953,7 +1989,160 @@ public IndexingPolicy build() {
+            if (0 <= reportMissingLimit) {
+                return false;
+            }
+            reportMissingLimit--;
+            return true;
+        }
+
+        boolean allowRepair() {
+            return allowRepair;
+        }
+
+        long getQuota() {
+            return quota;
+        }
+
+        /**
+         * Create an index scrubbing policy builder.
+         * @return a new {@link ScrubbingPolicy} builder
+         */
+        @Nonnull
+        public static Builder newBuilder() {
+            return new Builder();
+        }
+
+        /**
+         * Builder for {@link ScrubbingPolicy}.
+         *
+         * <pre><code>
+         * OnlineIndexer.ScrubbingPolicy.newBuilder().setReportMissingLimit(100).setScrubDangling(false).build()
+         * </code></pre>
+         *
+         */
+        @API(API.Status.UNSTABLE)
+        public static class Builder {
+            boolean scrubDangling = true;
+            boolean scrubMissing = true;
+            int reportDanglingLimit = 1000;
+            int reportMissingLimit = 1000;
+            boolean allowRepair = true;
+            long quota = 0;
+
+            protected Builder() {
+            }
+
+            /**
+             * If never called, the defaulted to allow scanning the index, looking for dangling index entries.
+             * @param scrubDangling if true
+             * @return this builder.
+             */
+            public Builder setScrubDangling(final boolean scrubDangling) {
+                this.scrubDangling = scrubDangling;
+                return this;
+            }
+
+            /**
+             * If never called, the defaulted to allow scanning the record store, looking for missing index entries.
+             * @param scrubMissing if true
+             * @return this builder.
+             */
+            public Builder setScrubMissing(final boolean scrubMissing) {
+                this.scrubMissing = scrubMissing;
+                return this;
+            }
+
+            /**
+             * If never called, the default is allowing (up to) 1000 error reports.
+             * @param reportDanglingLimit the max number of dangling-indexes errors to report.
+             * @return this builder.
+             */
+            public Builder setReportDanglingLimit(final int reportDanglingLimit) {
+                this.reportDanglingLimit = reportDanglingLimit;
+                return this;
+            }
+
+            /**
+             * If never called, the default is allowing (up to) 1000 error reports.
+             * @param reportMissingLimit the max number of missing-indexes errors to report.
+             * @return this builder.
+             */
+            public Builder setReportMissingLimit(final int reportMissingLimit) {
+                this.reportMissingLimit = reportMissingLimit;
+                return this;
+            }
+
+            /**
+             * Set 'scrub but do not repair' mode.
```

Comment:
I might rephrase this as "Set whether the index is repaired when scrubbing" or something like that. The way it's phrased makes it sound like if it's set to `true`, then it does _not_ repair. (This is clarified in the `param` description, but it should be a little more clear from the overall description, too.)

Reply:
Changed to `Set whether the scrubber, if finding an error, will repair it.`